File: app/src/main/java/de/exercicse/jrossbach/podcast/PodcastItemAdapter.java
package de.exercicse.jrossbach.podcast;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

/**
 * Created by jrossbach on 15.07.17.
 */

public class PodcastItemAdapter extends RecyclerView.Adapter<PodcastItemViewHolder> {


    private List<PodcastItemVieModel> podcastItemVieModelList;

    @Override
    public PodcastItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.podcast_item, parent, false);
        return new PodcastItemViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(PodcastItemViewHolder holder, int position) {
        holder.itemImageUrl = podcastItemVieModelList.get(position).imageUrl;
        holder.itemTitel = podcastItemVieModelList.get(position).title;
    }

    @Override
    public int getItemCount() {
        return podcastItemVieModelList.size();
    }

    public void setItems(List<PodcastItemVieModel> podcastItems){
        this.podcastItemVieModelList = podcastItems;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/de/exercicse/jrossbach/podcaster/feature/search/ui/adapter/PodcastItemAdapter.kt
package de.exercicse.jrossbach.podcaster.feature.search.ui.adapter

import android.view.ViewGroup
import de.exercicse.jrossbach.podcaster.base.ui.adapter.BaseAdapter
import de.exercicse.jrossbach.podcaster.feature.search.ui.PodCastItemClickListener
import de.exercicse.jrossbach.podcaster.feature.search.ui.PodcastItemViewModel

class PodcastItemAdapter(private val clickListener: PodCastItemClickListener) : BaseAdapter<PodcastItemViewModel, PodcastItemViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PodcastItemViewHolder {
        return PodcastItemViewHolder.create(parent, clickListener)
    }
}